pennylane_lightning/core/_measurements_base.py

```diff
@@ -197,16 +213,24 @@ def get_measurement_function(
         """
         if isinstance(measurementprocess, StateMeasurement):
             if isinstance(measurementprocess, ExpectationMP):
-                if isinstance(measurementprocess.obs, (qml.Identity, qml.Projector)):
-                    return self.state_diagonalizing_gates
+                if self._use_mpi:
+                    if isinstance(measurementprocess.obs, (qml.Projector)):
+                        return self.state_diagonalizing_gates
```

Comment:
`Codecov` complains

Reply:
Yes, we are not covered because we need to explore the support for `Projector` with MPI [sc-74357]